src/main/scala/edu/berkeley/ce/rockslicing/JointGenerator.scala

```diff
@@ -0,0 +1,210 @@
+    *         to generate full joint set.
+    */
+  private def findMasterJoints(globalOrigin: Array[Double], lowerLeftCorner: Array[Double],
+                               jointSets: Seq[JointSet]): Seq[Joint] = {
+    val masterJoints = jointSets flatMap { jointSet =>
+      if (!jointSet.stochasticFlag) {
+        // TODO: Implement non-persistent joint generator
+        val normal = findJointNormal(jointSet.strike, jointSet.dip)
+        Some(Joint(normal, globalOrigin, center=Array(lowerLeftCorner(0), lowerLeftCorner(1), lowerLeftCorner(2)),
+          phi = jointSet.phi, cohesion = jointSet.cohesion, shape=Vector.empty))
+      } else {
+        // TODO: Implement stochastic generator
+        throw new UnsupportedOperationException("ERROR: Joint Input Data: Stochastic joint generation " +
+          "not yet implemented")
+      }
+    }
+    masterJoints
+  }
+}
+
+/**
+  * Provides functionality to generate joint sets for input joint parameters and faces
+  * that represent faces of input rock volume.
+  *
+  * @param globalOrigin Coordinates of the global origin
+  * @param boundingBox Coordinates specifying the lower left and upper right corners of the box that bounds
+  *                    the domain of interest.
+  * @param rockVolumeData Seq of arrays describing each of the faces that define the boundaries of
+  *                       the rock volume of interest. The inputs in each array are strike, dip and the x,
+  *                       y and z coordinates of a point that lies within the plane containing the face.
+  *                       Strike and dip should be specified in degrees.
+  * @param jointSetData Seq of arrays containing the input data representing joint sets. The inputs in each array
+  *                     are strike, dip, joint spacing, persistence, phi, cohesion and optional stochastic parameters
+  */
+case class JointGenerator(globalOrigin: Array[Double], boundingBox: Array[Double], rockVolumeData: Seq[Array[Double]],
+                          jointSetData: Seq[JointSet]) {
+  val origin = globalOrigin
+  val lowerLeftCorner = Array(boundingBox(0), boundingBox(1), boundingBox(2))
+  val upperRightCorner = Array(boundingBox(3), boundingBox(4), boundingBox(5))
+  val rockVolume = JointGenerator.findBoundingFaces(globalOrigin, rockVolumeData)
+  val masterJoints = JointGenerator.findMasterJoints(globalOrigin, lowerLeftCorner, jointSetData)
+  val jointSets = generateJointSets(jointSetData)
+
+  /**
+    * Generates joint sets for each of the input joint sets
+    *
+    * @param jointSetData Seq of JointSets containing the input data representing joint sets. The inputs in each array
+    *                     are strike, dip, joint spacing, persistence, phi, cohesion and optional stochastic parameters
+    * @return Seq of joints for each input joint set
+    */
+  def generateJointSets(jointSetData: Seq[JointSet]): Seq[Seq[Joint]] = {
```

Comment:
It appears that the only place this method is called is a few lines above. Is there any reason not to make it `private` since it's only used within this class?